lib/livebook/runtime/erl_dist/runtime_server.ex

```diff
@@ -177,12 +179,22 @@ defmodule Livebook.Runtime.ErlDist.RuntimeServer do
     end
   end
 
+  def handle_info({:evaluation_finished, _ref}, state) do
+    send_memory_usage(state)
```

Comment:
Since we are already keeping the timer we can do `Process.cancel_timer(state.memory_timer_ref)` and then schedule a new one. This way we always send 15 seconds after last schedule or last evaluation :)

Reply:
There is a race condition here, unfortunately. When you cancel the timer, the message may have already been delivered. Which means you will schedule a new one and the message will be executed and still re-run it. So if you cancel it, you need to prune the inbox. I am thinking it is better to not cancel altogether.

Reply:
Good catch, addressed in #916!